--- src/log.c	2014-02-02 15:41:29.000000000 -0800
+++ /tmp/log.c	2014-04-18 22:02:21.914798083 -0700
@@ -795,6 +795,16 @@
 	int nblogger;
 	char *log_ptr;
 
+	FILE *file; 
+	static char* logfile=0;
+	if (!logfile) {
+		logfile = getenv("HAPROXY_LOGFILE");
+		if (logfile)
+			printf("Logging to file %s because HAPROXY_LOGFILE is set\n", logfile);
+		else
+			printf("HAPROXY_LOGFILE is not set\n");
+	}
+
 	dataptr = message;
 
 	if (p == NULL) {
@@ -870,15 +880,21 @@
 			log_ptr--;
 		} while (fac_level && log_ptr > dataptr);
 		*log_ptr = '<';
-
-		sent = sendto(*plogfd, log_ptr, size - (log_ptr - dataptr),
+ 
+                if (logfile) {
+                        file = fopen(logfile,"a");
+                        fprintf(file, "%s", log_ptr); 
+                        fclose(file);
+                } else {
+	    	        sent = sendto(*plogfd, log_ptr, size - (log_ptr - dataptr),
 			      MSG_DONTWAIT | MSG_NOSIGNAL,
 			      (struct sockaddr *)&logsrv->addr, get_addr_len(&logsrv->addr));
-		if (sent < 0) {
-			Alert("sendto logger #%d failed: %s (errno=%d)\n",
-				nblogger, strerror(errno), errno);
-		}
-		nblogger++;
+		        if (sent < 0) {
+		          	Alert("sendto logger #%d failed: %s (errno=%d)\n",
+			       	nblogger, strerror(errno), errno);
+		        }
+               }
+	       nblogger++;
 	}
 }
 
